league.js

```diff
@@ -0,0 +1,167 @@
+            throw new Error("findPairing requires at least one username.");
+        }
+        black = black || undefined;
+        var possibilities = this._pairings;
+        if (white) {
+            possibilities = _.filter(possibilities, function(item) {
+                return (
+                    item.white.toLowerCase().includes(white) ||
+                    item.black.toLowerCase().includes(white)
+                );
+            });
+
+        }
+        if (black) {
+            possibilities = _.filter(possibilities, function(item) {
+                return (
+                    item.white.toLowerCase().includes(black) ||
+                    item.black.toLowerCase().includes(black)
+                );
+            });
+        }
+        return possibilities;
+    },
+
+    //--------------------------------------------------------------------------
+    // Figures out the current scheduling information for the round.
+    //--------------------------------------------------------------------------
+    'getCurrentRoundSchedule': function(callback) {
+        var query_options = {
+            'min-row': 1,
+            'max-row': 100,
+            'min-col': 1,
+            'max-col': 8,
+            'return-empty': true
+        }
+        var self = this;
+        spreadsheets.get_rows(
+            self.options.spreadsheet.service_account_auth,
+            self.options.spreadsheet.key,
+            query_options,
+            function(err, rows) {
+                if (err) { return callback(err, rows); }
+                var new_pairings = [];
+                rows.forEach(function(row) {
+                    if (!row['white'].value || !row['black'].value) { return; }
+                    var link = spreadsheets.parse_hyperlink(row['result'].formula || "");
+                    var date_string = row[self.options.spreadsheet.schedule_colname].value || '';
+                    date_string = date_string.trim()
+                    var date = moment.utc(
+                        moment.utc().year() + "/" + date_string,
+                        "YYYY/MM/DD @ HH:mm",
+                        true
+                    );
+                    if (!date.isValid()) {
+                        date = undefined;
+                    }
+                    new_pairings.push({
+                        white: self.canonical_username(row['white'].value),
+                        black: self.canonical_username(row['black'].value),
+                        result: link['text'],
+                        url: link['href'],
+                        scheduled_date: date
+                    });
+                });
+                self._pairings = new_pairings;
+                callback(undefined, self._pairings);
+            }
+        );
+    }
+};
+
+function League(options) {
+    this.options = {};
+    _.extend(this.options, LEAGUE_DEFAULTS, options || {});
+    _.extend(this, league_attributes);
+};
+
+var _cache = {};
```

Comment:
I suggest moving cache into the getLeague function to avoid scope pollution
```
function getLeague(league_name, config) {
    this._cache = ( this.cache ? this.cache : {} );

    if(!this._cache){
        console.log(league_name);

        // Else create it if there is a config for it.
        var league_options = config[league_name] || undefined;
        if (league_options) {
            this._cache[league_name] = new League(league_options);
        }else{
            console.log("Couldn't find options for " + league_name + " league. Not creating object.");
        }
    }
    return this._cache[league_name];
}
```

Also a little refactoring.